### ISSUE TITLE
Collection expressions: inapplicable overloads considered in overload resolution

### ISSUE THREAD

**Issue by cston** [dotnet/roslyn#69634]
**Version Used**: d838fac24ed353b2002e0bacdd1df5c66a61de04

**Steps to Reproduce**:

`F([1, 2, 3])` below is reported as ambiguous, even though only one overload is applicable.

```csharp
class Program
{
    static void F(string[] arg) { }

    static void F(int[] arg) { }

    static void Main()
    {
        F([1, 2, 3]); // error CS0121: ambiguous between 'F(string[])' and 'F(int[])'
    }
}
```

**Expected Behavior**:
Compiles successfully.

Relates to test plan https://github.com/dotnet/roslyn/issues/66418

**Actual Behavior**:
Error reported:  `error CS0121: ambiguous between 'F(string[])' and 'F(int[])'`

**Comment by cston** [comment]
Fixed in https://github.com/dotnet/roslyn/pull/69627.
